server/api/crud/model_monitoring/deployment.py

```diff
@@ -875,6 +882,212 @@ async def delete_monitoring_function(
+
+        credentials_dict = {
+            key: server.api.crud.Secrets().get_project_secret(
+                project=self.project,
+                provider=mlrun.common.schemas.SecretProviderName.kubernetes,
+                secret_key=key,
+                allow_secrets_from_k8s=True,
+            )
+            for key in mlrun.common.schemas.model_monitoring.ProjectSecretKeys.mandatory_secrets()
+        }
+        # check if all the credentials are set (stream is not mandatory for now for BC, Todo: del in 1.9.0)
+        if all(
+            [
+                val is not None
+                if key
+                != mlrun.common.schemas.model_monitoring.ProjectSecretKeys.STREAM_PATH
+                else True
+                for key, val in credentials_dict.items()
+            ]
+        ):
+            return
+
+        if not only_project_secrets:  # for upgrade case
+            need_to_set = False
+            try:
+                server.api.crud.Functions().get_function(
+                    name=mm_constants.MonitoringFunctionNames.STREAM,
+                    db_session=self.db_session,
+                    project=self.project,
+                )
+                need_to_set = True
+            except mlrun.errors.MLRunNotFoundError:
+                if any(
+                    [val is not None for val in credentials_dict.values()]
+                ):  # check if one of the cred is already set
+                    need_to_set = True
+            finally:
+                if need_to_set and None in credentials_dict.values():
+                    self.set_credentials(
+                        endpoint_store_connection=credentials_dict.get(
+                            mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ENDPOINT_STORE_CONNECTION
+                        ),
+                        stream_path=credentials_dict.get(
+                            mlrun.common.schemas.model_monitoring.ProjectSecretKeys.STREAM_PATH
+                        ),
+                        tsdb_connection=credentials_dict.get(
+                            mlrun.common.schemas.model_monitoring.ProjectSecretKeys.TSDB_CONNECTION
+                        ),
+                        default_secrets="v3io",
+                    )
+
+        raise mlrun.errors.MLRunBadRequestError(
+            "Model monitoring credentials are not set.\n"
+            "Please set them using the set_model_monitoring_credentials API/SDK "
+            "or pass fetch_credentials_from_sys_config=True when using enable_model_monitoring API/SDK."
+        )
+
+    def set_credentials(
+        self,
+        access_key: typing.Optional[str] = None,
+        endpoint_store_connection: typing.Optional[str] = None,
+        stream_path: typing.Optional[str] = None,
+        tsdb_connection: typing.Optional[str] = None,
+        default_secrets: typing.Optional[str] = "",  # used only for upgrade case
```

Comment:
as for the default_secrets, I recommend to convert it to a `dict` that represents secret key-value. Then, it actually supports multiple optional keys.